File: booksources/examples/mpi/p/hello.py
#!/usr/bin/env python

################################################################
################################################################
####
#### This program file is part of the book and course
#### "Parallel Computing"
#### by Victor Eijkhout, copyright 2013-5
####
#### hello.py : just init and hello world
####
################################################################
################################################################

from mpi4py import MPI

comm = MPI.COMM_WORLD

ntids = comm.Get_size()
mytid = comm.Get_rank()

if mytid==0:
    print "Starting %d processes" % ntids

File: booksources/examples/mpi/c/hello.c
/* %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   %%%%
   %%%% This program file is part of the book and course
   %%%% "Parallel Computing"
   %%%% by Victor Eijkhout, copyright 2013-6
   %%%%
   %%%% hello.c : simple hello world
   %%%%
   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
*/

#include <stdlib.h>
#include <stdio.h>
#include <mpi.h>

int main(int argc,char ** argv) {

#include "globalinit.c"

  printf("processor [%d] says hello!\n",mytid);

  MPI_Finalize();
  return 0;
}
